### PR TITLE
Adding urls

### PR DESCRIPTION
**Pull request by vallettea**

### CODE REVIEW

File: src/views/layout.jsx [hunk 1]
```diff
@@ -24,208 +25,216 @@ var injectTapEventPlugin = require("react-tap-event-plugin");
...
-	},
-	render: function() {
+        search(data)
+        .then(function(result){
 
-		// If a point has been selected, we popup a detailed sheet
-		var showDetail = (this.state.detailedObject !== null &&
-			this.state.detailedObject !== undefined);
+            // window.location = "?" + queryString.stringify(parameters.geoloc);
+            window.history.pushState("object or string", "Title", "?" + queryString.stringify(parameters.geoloc));
```
Comment (DavidBruant) [inline src/views/layout.jsx]: Use `page(path)` here instead. See https://visionmedia.github.io/page.js/

Comment (vallettea) [inline src/views/layout.jsx]: ah ok starting to understand...

**Comment by vallettea** [comment]
this is what to be done:

- [ ] adapt the map when url contains a bounding box
- [ ] two calls of the route / after search
- [ ] the points are not loaded afet zoom or pan when starting from the url
- [ ] add 6element logo that point to /
- [ ] once a detailed view is selected the go back arrow works but the url is wrong
- [ ] once a url of detailed place has been call all routes point to it always until the server is restarted

File: server/index.js [hunk 2]
```diff
@@ -86,6 +88,42 @@ app.get('/', function(req, res){
...
+            .then(function(geoJson){
+            
+                layoutData.detailedObject = geoJson[0];
+
+                // Material-ui needs to change the global.navigator.userAgent property
+                // (just during the React  rendering)
+                // Waiting for Material team to fix it 
+                //https://github.com/callemall/material-ui/pull/2007#issuecomment-155414926
+                global.navigator = {'userAgent': req.headers['user-agent']};
+                renderDocumentWithData(doc, layoutData, Layout);
```
Comment (DavidBruant) [inline server/index.js]: A first problem we should be taking care of is that both `/` and `/place/:placeId` are rendered using the same React component (`Layout`) although they should render something different.

`Layout` most renders something different based on the props, but this is not really scalable and/or maintainable.
The approach I've been using in MyWI is to have as many top-level components as I have routes and prefix them with `-Screen`. See https://github.com/MyWebIntelligence/MyWebIntelligence/tree/master/client/components (`OraclesScreen`, `TerritoireListScreen`, `TerritoireViewScreen`). This way, it's crystal-clear that there is a bijection ( ;-) ) between routes and React "screen components" and the bug we're encountering right now (mismatch between the URL and what's being displayed) should be impossible by design.

Obviously, inside these top-level components, the goal is to reuse as many smaller components as possible (headers, etc.).

In our case, we probably should be having something like `MainScreen` (for `/`) and `PlaceScreen` (for `/place/:placeId`)

**Pull request merged**